docs/source/pjsua2/using/media_video.rst

```diff
@@ -0,0 +1,293 @@
+destinations, the bridge will take care of duplicating the video frame from the source to the 
+multiple destinations. The bridge will even take care of mixing video with different sampling
+rates.
+
+In PJSUA2, all video media objects, of class :cpp:class:`pj::VideoMedia`, are registered to
+the central conference bridge for easier manipulation. At first, a registered video media will
+not be connected to anything, so media will not flow from/to any objects. An video media source
+can start/stop the transmission to a destination by using the API
+:cpp:func:`pj::VideoMedia::startTransmit()` and :cpp:func:`pj::VideoMedia::stopTransmit()`.
+
+.. note::
+
+    An video media object registered to the conference bridge will be given a port ID number that 
+    identifies the object in the bridge. Application can use the API :cpp:func:`pj::VideoMedia::getPortId()` 
+    to retrieve the port ID. Normally, application should not need to worry about the conference 
+    bridge and its port ID (as all will be taken care of by the ``Media`` class) unless application 
+    wants to create its own custom video media.
+
+    In the video conference bridge, the port zero is not special like in audio, which is designated
+    for the main audio device. In video, port zero can be assigned to any type of video object.
+
+
+Starting camera preview
+----------------------------
+Application can start the camera (or any capture device in general) preview using video object
+:cpp:class:`pj::VideoPreview`.
+
+.. note::
+    Application does not need to start a camera preview manually to setup a video call,
+    the camera preview will be started automatically once the video call is established,
+    it is just that by default the video preview windows is hidden. The capture device
+    to be used in a video call is configurable in account setting.
+
+.. code-block:: c++
+
+    void StartPreview(int device_id, void* hwnd, int width, int height, int fps)
+    {
+        try {
+            // Set the video capture device format.
+	    VidDevManager &mgr = Endpoint::instance().vidDevManager();
+            MediaFormatVideo format = mgr.getFormat(device_id);
+	    format.width    = width;
+	    format.height   = height;
+	    format.fpsNum   = fps;
+	    format.fpsDenum = 1;
+            mgr.setFormat(device_id, format, true);
+        
+	    // Start the preview on a panel with window handle 'hwnd'.
+	    // Note that if hwnd is set to NULL, library will automatically create
+	    // a new floating window for the rendering.
+	    VideoPreviewOpParam param;
+	    param.window.handle.window = (void*) hwnd;
+	    
+	    VideoPreview preview(device_id);
+	    preview.start(param);
+        } catch(Error& err) {
+        }
+    }
+
+See :cpp:class:`pj::VideoPreview`, :cpp:class:`VideoPreviewOpParam`, :cpp:class:`MediaFormatVideo`,
+and :cpp:func:`pj::Endpoint::vidDevManager()` for reference.
+
+.. note::
```

Comment:
Since this section is important, it's probably need to be highlighted more than a small note. Perhaps a separate section such as "Important note about threading."

Here's a reference from Apple's official doc that can be useful here:
https://developer.apple.com/documentation/xcode/improving-app-responsiveness?changes=_1
Although it is written by Apple, I believe it's applicable in general, i.e. in order to improve app responsiveness and user experience, the doc recommends to `keep the UI/main thread free from non-UI work.`

So in the section, it should contain the same recommendation, i.e. to avoid calling PJSIP API from the UI thread since:
- it may take some time to complete, or
- it may block while trying to acquire a lock